File: runtime/src/de/fatox/meta/api/graphics/ShaderInfo.java
package de.fatox.meta.api.graphics;

import com.badlogic.gdx.graphics.g3d.Shader;

public interface ShaderInfo {
    class RenderTarget {
        public final String name;

        public RenderTarget(String name) {
            this.name = name;
        }
    }

    RenderTarget[] getRenderTargets();

    Shader getShader();

    String getName();
}

File: runtime/src/de/fatox/meta/api/graphics/ShaderInfo.kt
package de.fatox.meta.api.graphics

import com.badlogic.gdx.graphics.g3d.Shader

interface ShaderInfo {
	val isDepth: Boolean

	val renderTargets: Array<RenderTarget>

	val shader: Shader

	val name: String

	class RenderTarget(val name: String)
}
